File: bronze/5073/c++/main.cpp
#include <iostream>

int main()
{
    int a, b, c, sum, target;

    while (1) {
        std::cin >> a >> b >> c;
        if (!a && !b && !c) {
            break;
        }
        sum = 0;
        target = 0;


        if (a > b && a > c) {
            target = a;
            sum = b + c;
        } else if (b > a && b > c) {
            target = b;
            sum = a + c;
        } else {
            target = c;
            sum = a + b;
        }
        if (target >= sum) {
            std::cout << "Invalid\n";
            continue;
        }


        if (a == b && a == c && b == c) {
            std::cout << "Equilateral\n";
        } else if (a == b || a == c || b == c) {
            std::cout << "Isosceles\n";
        } else {
            std::cout << "Scalene\n";
        }
    }


    return 0;
}

File: bronze/5073/python/main.py

if (__name__ == "__main__"):
    while (1):
        a, b, c = map(int, input().split())
        if (a == 0 and b == 0 and c == 0): break
        
        sum, target = (0, 0)
        target = max([a, b, c])
        if (target == a): sum = b + c
        elif (target == b): sum = a + c
        elif (target == c): sum = a + b
        if (target >= sum): print("Invalid"); continue

        if (a == b and a == c and b == c):
            print("Equilateral")
        elif (a == b or a == c or b == c):
            print("Isosceles")
        else :
            print("Scalene")